app/gui/src/project-view/components/GraphEditor/widgets/WidgetPort.vue

```diff
@@ -143,6 +143,13 @@ export const widgetDefinition = defineWidget(
     score: (props, _db) => {
       const portInfo = injectPortInfo(true)
       const value = props.input.value
+      // This is a workaround to avoid automatic port around type annotated expressions
+      // in argument positions. A port needs to be created with portId override, and it
+      // will be handled by `WidgetTypeCastPort`. Without this check, the port will be
+      // created with invalid portId because of `ArgumentInfoKey` being set on the input.
+      if (props.input.portId.includes('TypeAnnotated')) {
```

Comment:
I prefer to avoid inspecting an AstId--they contain the type name to aid debugging, but they are intended to be opaque values. In the future (#10752) we may change the format.
Maybe here we could check if the input is an `Ast`, and then if the `portId` doesn't match the `AstId` we don't create a port unless `forcePort` is set?

Reply:
Yeah, it’s definitely an ugly hack, but I’m struggling to find a solution. The issue here is that `portId` **equals** `AstId` at this moment, we change it only in `WidgetTypeCast`, which has lower priority, so it matches later. 
I’m not sure there is some order of priorities in which it starts to work without breaking some cases. The only way to know that we don’t want to create a port immediately is to inspect the AST and see that it is actually a TypeAnnotated, but I don’t want to do 
It looks to me that we need some kind of bottom-up propagation of information through the tree to implement it correctly.

Although I found a better check, without comparing with the string:
```
props.input.value instanceof Ast.TypeAnnotated && props.input.value.id === props.input.portId
```